src/app/patients/[patientId]/appointments/page.tsx

```diff
@@ -1,10 +1,163 @@
+        ._include("Appointment", "actor"),
+    upcomingAppointmentsSearchController.pageUrl,
+  );
+
+  const pastAppointmentsSearchController =
+    useFhirSearchControllerNext<AppointmentSortOrder>("past", {
+      defaultSort: "-date",
+      pageSize: 10,
+    });
+  const pastAppointmentsQuery = useFhirSearch(
+    "Appointment",
+    (search) =>
+      search
+        .patient(patient.id)
+        .date(currentDate.toDateString(), "lt")
+        ._total("accurate")
+        ._count(pastAppointmentsSearchController.pageSize)
+        ._sort(pastAppointmentsSearchController.sort)
+        ._include("Appointment", "actor"),
+    pastAppointmentsSearchController.pageUrl,
+  );
+
+  return (
+    <Stack>
+      <AppointmentTable
+        query={upcomingAppointmentsQuery}
+        searchController={upcomingAppointmentsSearchController}
+        title="Upcoming Appointments"
+      />
+      <AppointmentTable
+        query={pastAppointmentsQuery}
+        searchController={pastAppointmentsSearchController}
+        title="Past Appointments"
+      />
+    </Stack>
+  );
+}
+
+function AppointmentTable({
+  query,
+  searchController,
+  title,
+}: {
+  query: UseQueryResult<BundleNavigator<Retrieved<Appointment>>>;
+  searchController: UseFhirSearchControllerValue<AppointmentSortOrder>;
+  title: string;
+}) {
   return (
     <Paper>
-      <Title order={3}>Appointments</Title>
+      <Title order={3}>{title}</Title>
+      <FhirQueryLoader query={query}>
+        <FhirTable
+          {...query}
+          {...searchController}
+          columns={[
+            {
+              key: "appointmentType",
+              title: "Appointment Type",
+              render: (appointment: Appointment) => (
+                <FhirValue
+                  type="CodeableConcept"
+                  value={appointment.appointmentType}
+                />
+              ),
+            },
+            {
+              key: "date",
+              title: "Scheduled Time",
+              sortable: true,
+              render: (appointment: Appointment) => (
+                <FhirValue type="dateTime" value={appointment.start} />
+              ),
+            },
+            {
+              key: "status",
+              title: "Status",
+              render: (appointment: Appointment) => (
+                <FhirValue type="string" value={appointment.status} />
+              ),
+            },
+            {
+              key: "participant",
+              title: "Provider",
+              render: (appointment: Appointment) => (
+                <FhirValue
+                  type="HumanName"
+                  value={
+                    (
+                      appointment.participant
+                        .find((participant: AppointmentParticipant) =>
+                          isReferenceOf(participant.actor, "Practitioner"),
+                        )
+                        // @ts-ignore
+                        ?.actor?.included() as Practitioner
```

Comment:
@julienblin This is still working but .included() is giving me a typer error. Not sure what the typing fix is here.